gremlin-dotnet/src/Gremlin.Net/Process/Traversal/Lambda.cs

```diff
@@ -0,0 +1,59 @@
+﻿#region License
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#endregion
+
+namespace Gremlin.Net.Process.Traversal
+{
+    /// <summary>
+    ///     Represents a lambda.
+    /// </summary>
+    public interface ILambda : IFunction, IBiFunction, IPredicate, IUnaryOperator, IBinaryOperator, IComparator,
+        IConsumer, ISupplier
+    {
```

Comment:
We could add the properties to the `ILambda` interface, in the same way it's done in java: https://github.com/apache/tinkerpop/blob/f5f4a046f211851e901d303c6593975314171e55/gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/util/function/Lambda.java#L33-L39

The risk of that interface ever changing is shared with Java and a single serializer (for any `ILambda`) will only be needed.